--- a/iavl/store.go
+++ b/iavl/store.go
@@ -57,7 +57,8 @@ func LoadStoreWithOpts(db dbm.DB, logger log.Logger, key types.StoreKey, id type
 	if initialVersion == 0 {
 		initialVersion = 1
 	}
-	opts = append(opts, iavl.InitialVersionOption(initialVersion))
+	opts = append(opts, iavl.InitialVersionOption(initialVersion),
+		iavl.HasherOptionForStore(key.Name()))
 	tree := iavl.NewMutableTree(wrapper.NewDBWrapper(db), cacheSize, disableFastNode, logger, opts...)
 
 	isUpgradeable, err := tree.IsUpgradeable()
